test/src/e2e_vm_tests/test_programs/should_pass/language/addrof_intrinsic/src/main.sw

```diff
@@ -1,13 +1,6 @@
 script;
 
-const B1 = Address {
-    value: 0x0100000000000000000000000000000000000000000000000000000000000010
-};
-
-pub fn addr_of<T>(val: T) -> raw_ptr {
-    if !__is_reference_type::<T>() {
-        revert(0);
-    }
+pub fn as_raw_ptr<T>(val: T) -> raw_ptr {
```

Comment:
Having the argument being `&T` brings more strictness here. And for the `as_raw_ptr(0)` case, it is more idiomatic if `is_null()` is used as suggested below. 
```suggestion
pub fn as_raw_ptr<T>(val: &T) -> raw_ptr {
```